Code/Tools/RemoteConsole/Core/RemoteConsoleCore.cpp

```diff
@@ -306,15 +324,14 @@ bool SRemoteServer::ReadBuffer(const char* buffer, int data)
     {
         // Create the event from the current sub string in the buffer.
         IRemoteEvent* event = SRemoteEventFactory::GetInst()->CreateEventFromBuffer(curBuffer, bytesRemaining);
-        
+
         result &= (event != nullptr);
         if (event)
         {
             if (event->GetType() != eCET_Noop)
             {
-                m_lock.Lock();
+                AZStd::lock_guard<AZStd::recursive_mutex> lock(m_mutex);
```

Comment:
repeat: Can be replaced with

```suggestion
                AZStd::scoped_lock lock(m_mutex);
```